src/test/java/address/unittests/browser/HyperBrowserTest.java

```diff
@@ -42,14 +43,14 @@ public HyperBrowserTest() throws MalformedURLException {
 
     @Test
     public void testFullFeatureBrowser_loadUrl_urlAssigned() throws MalformedURLException, InterruptedException {
-        HyperBrowser browser = new HyperBrowser(HyperBrowser.Type.FULL_FEATURE_BROWSER, 1, Optional.empty());
+        HyperBrowser browser = new HyperBrowser(new EmbeddedBrowserFactory(EmbeddedBrowserFactory.Type.FULL_FEATURE_BROWSER), 1, Optional.empty());
         Page page = browser.loadUrl(new URL("https://github.com"));
         assertTrue(UrlUtil.compareBaseUrls(page.getBrowser().getOriginUrl(), new URL("https://github.com")));
     }
 
     @Test
     public void testFullFeatureBrowser_loadUrls_urlsAssigned() throws MalformedURLException, IllegalAccessException, NoSuchFieldException, InterruptedException {
-        HyperBrowser browser = new HyperBrowser(HyperBrowser.Type.FULL_FEATURE_BROWSER, 3, Optional.empty());
+        HyperBrowser browser = new HyperBrowser(new EmbeddedBrowserFactory(EmbeddedBrowserFactory.Type.FULL_FEATURE_BROWSER), 3, Optional.empty());
```

Comment:
We are trying to abstract away the concrete implementation of EmbeddedBrowser and not HyperBrowser?